test/lib/jdk/test/lib/util/FileUtils.java

```diff
@@ -377,6 +377,35 @@ public static long getProcessHandleCount() {
         }
     }
 
+    /**
+     * Patches a part of a file.
+     * @param path of file
+     * @param fromLine the first line to patch. This is the number you see in an editor, 1-based, inclusive.
+     * @param toLine the last line to patch. This is the number you see in an editor, inclusive.
+     * @param from the lines to remove, used to ensure the correct lines are removed. Ignored if null.
+     * @param to the newly added line, can be multiple lines or empty. Cannot be null.
+     * @throws IOException
+     */
+    public static void patch(Path path, int fromLine, int toLine, String from, String to) throws IOException {
+        var lines = Files.readAllLines(path);
+        if (from != null) {
+            var removed = "";
+            for (int i = fromLine; i <= toLine; i++) {
+                removed += lines.remove(fromLine - 1).trim();
```

Comment:
Also calling trim() assumes that white spaces are not significant. This might not be the case in the general case (for instance - think of markdown files were leading spaces are significant).

Reply:
The comparison is intentionally made lax so the caller does not need to provide the exact indentation or even new line characters. We think along with `fromLine` and `toLine` this is enough to make sure we are not modifying the wrong lines.

Reply:
We can provide a more sophisticated `Function<String,String>` replacer we want to let user customize all the details. This time we still only want them to be string literals. I agree we can keep the new lines inside, but trimming on each line and the final block is still useful so caller does not need to care about indentation and empty lines at both ends.

Reply:
OK - if you keep the internal new  lines I have no objection. The API doc should however say that lines will be trimmed before comparing them.